api/segments/serializers.py

```diff
@@ -120,6 +126,22 @@ def _update_or_create_segment_rules(
                 child_rules, rule=child_rule, is_create=is_create
             )
 
+    def _update_or_create_metadata(self, metadata_data, segment=None):
+        for metadata_item in metadata_data:
+            metadata_id = metadata_item.pop("id", None)
+            if metadata_item.get("delete"):
+                Metadata.objects.filter(id=metadata_id).delete()
+                continue
```

Comment:
I'm a bit confused here - I don't think this is how it works for other entities?

Reply:
I think that this was answered here:  https://github.com/Flagsmith/flagsmith/pull/3315